### ISSUE TITLE
check-multi-machine fails to navigate to /playground/test on second machine

### ISSUE THREAD

**Issue by stefwalter** [cockpit-project/cockpit#2648]
check-multi-machine fails to navigate to /playground/test on secondary machine:

```
Wrote check-multi-machine-testFrameNavigation-FAIL.png
Journal database copied to check-multi-machine-testFrameNavigation-10.111.111.11-FAIL.journal
Journal database copied to check-multi-machine-testFrameNavigation-10.111.111.13-FAIL.journal
Journal database copied to check-multi-machine-testFrameNavigation-10.111.111.12-FAIL.journal
..> couldn't load os data: cancelled
> problem retrieving real time metrics data: authentication-failed
> problem retrieving real time metrics data: authentication-failed
> problem retrieving real time metrics data: authentication-failed
> problem retrieving real time metrics data: authentication-failed
EWarning: Permanently added '10.111.111.11' (ECDSA) to the list of known hosts.
Warning: Permanently added '10.111.111.13' (ECDSA) to the list of known hosts.
Warning: Permanently added '10.111.111.12' (ECDSA) to the list of known hosts.

======================================================================
ERROR: testFrameNavigation (__main__.TestMultiMachine)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./check-multi-machine", line 231, in testFrameNavigation
    b.enter_page("/playground/test", m2.address)
  File "/home/hubbot/hubbot/3e0abb2a7cf3a663d261bab5becda05e5f9eadf2_r7_x86-64/test/testlib.py", line 338, in enter_page
    self.wait_present("iframe.container-frame[name='%s'][data-loaded]" % frame)
  File "/home/hubbot/hubbot/3e0abb2a7cf3a663d261bab5becda05e5f9eadf2_r7_x86-64/test/testlib.py", line 229, in wait_present
    return self.wait_js_func('ph_is_present', selector)
  File "/home/hubbot/hubbot/3e0abb2a7cf3a663d261bab5becda05e5f9eadf2_r7_x86-64/test/testlib.py", line 223, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))), timeout=self.phantom_wait_timeout)
  File "/home/hubbot/hubbot/3e0abb2a7cf3a663d261bab5becda05e5f9eadf2_r7_x86-64/test/testlib.py", line 647, in wait
    return self.run({'cmd': 'wait', 'cond': cond, 'timeout': timeout*1000})
  File "/home/hubbot/hubbot/3e0abb2a7cf3a663d261bab5becda05e5f9eadf2_r7_x86-64/test/testlib.py", line 610, in run
    raise Error("timeout")
Error: timeout
```

Seen here: http://files.cockpit-project.org/hubbot/3e0abb2a7cf3a663d261bab5becda05e5f9eadf2_r7_x86-64/hubbot.html

**Comment by stefwalter** [comment]
![check-multi-machine-testframenavigation-fail](https://cloud.githubusercontent.com/assets/795070/9542586/5eb0b18e-4d72-11e5-9f5e-8640f659071f.png)

**Comment by stefwalter** [comment]
Slightly different, but probably the same:

```
OK
Wrote check-multi-machine-testFrameNavigation-FAIL.png
Journal database copied to check-multi-machine-testFrameNavigation-10.111.111.11-FAIL.journal
Journal database copied to check-multi-machine-testFrameNavigation-10.111.111.13-FAIL.journal
Journal database copied to check-multi-machine-testFrameNavigation-10.111.111.14-FAIL.journal
..> couldn't load os data: cancelled
> problem retrieving real time metrics data: authentication-failed
> problem retrieving real time metrics data: authentication-failed
> problem retrieving real time metrics data: authentication-failed
> problem retrieving real time metrics data: authentication-failed
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
Page error: iframe.container-frame[name='10.111.111.13/playground/test'] not found
EWarning: Permanently added '10.111.111.11' (ECDSA) to the list of known hosts.
Warning: Permanently added '10.111.111.13' (ECDSA) to the list of known hosts.
Warning: Permanently added '10.111.111.14' (ECDSA) to the list of known hosts.

======================================================================
ERROR: testFrameNavigation (__main__.TestMultiMachine)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./check-multi-machine", line 231, in testFrameNavigation
    b.enter_page("/playground/test", m2.address)
  File "/home/hubbot/hubbot/cd157cf4a867eed463ddf6621370a7d416c705eb_r7_x86-64/test/testlib.py", line 339, in enter_page
    self.wait_visible("iframe.container-frame[name='%s']" % frame)
  File "/home/hubbot/hubbot/cd157cf4a867eed463ddf6621370a7d416c705eb_r7_x86-64/test/testlib.py", line 235, in wait_visible
    return self.wait_js_func('ph_is_visible', selector)
  File "/home/hubbot/hubbot/cd157cf4a867eed463ddf6621370a7d416c705eb_r7_x86-64/test/testlib.py", line 223, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))), timeout=self.phantom_wait_timeout)
  File "/home/hubbot/hubbot/cd157cf4a867eed463ddf6621370a7d416c705eb_r7_x86-64/test/testlib.py", line 647, in wait
    return self.run({'cmd': 'wait', 'cond': cond, 'timeout': timeout*1000})
  File "/home/hubbot/hubbot/cd157cf4a867eed463ddf6621370a7d416c705eb_r7_x86-64/test/testlib.py", line 608, in run
    raise Error(res['error'])
Error: iframe.container-frame[name='10.111.111.13/playground/test'] not found

----------------------------------------------------------------------
Ran 3 tests in 297.477s
```

http://files.cockpit-project.org/hubbot/cd157cf4a867eed463ddf6621370a7d416c705eb_r7_x86-64/hubbot.html

![](http://files.cockpit-project.org/hubbot/cd157cf4a867eed463ddf6621370a7d416c705eb_r7_x86-64/test/check-multi-machine-testFrameNavigation-FAIL.png)

**Comment by stefwalter** [comment]
Another case: http://files.cockpit-project.org/hubbot/aabe4811cb7b95872af53eb70d937c70370dc968_r7_x86-64/hubbot.html

**Comment by stefwalter** [comment]
Unfortunately it doesn't look like #2650 fixed the issue. Or at least not all the issues:

http://files.cockpit-project.org/hubbot/34a72ac463ba36c9b8947c226186045c68c5e697_r7_x86-64/hubbot.html

**Comment by stefwalter** [comment]
Is this the same issue? I can see it on my machine. It's a different test:

```
Wrote check-multi-machine-key-testBasic-FAIL.png
Journal database copied to check-multi-machine-key-testBasic-10.111.111.13-FAIL.journal
Journal database copied to check-multi-machine-key-testBasic-10.111.111.16-FAIL.journal
> transport closed, dropped message:  
{"problem":"cancelled","command":"close","channel":"1:1!18"}
> transport closed, dropped message:  
{"command":"close","channel":"1:1!21"}
> transport closed, dropped message:  
{"command":"close","channel":"1:1!22"}
> transport closed, dropped message:  
{"command":"close","channel":"1:1!26"}
> transport closed, dropped message:  
{"command":"close","channel":"1:1!27"}
> transport closed, dropped message:  
{"command":"kill","group":"localhost/shell/shell"}
> problem retrieving real time metrics data: authentication-failed
> problem retrieving real time metrics data: authentication-failed
> problem retrieving real time metrics data: authentication-failed
> problem retrieving real time metrics data: authentication-failed
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
Page error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found
EWarning: Permanently added '10.111.111.13' (ECDSA) to the list of known hosts.
Warning: Permanently added '10.111.111.16' (ECDSA) to the list of known hosts.

======================================================================
ERROR: testBasic (__main__.TestMultiMachineKeyAuth)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./check-multi-machine-key", line 166, in testBasic
    b.enter_page("server", m2.address)
  File "/data/src/cockpit/test/testlib.py", line 339, in enter_page
    self.wait_visible("iframe.container-frame[name='%s']" % frame)
  File "/data/src/cockpit/test/testlib.py", line 235, in wait_visible
    return self.wait_js_func('ph_is_visible', selector)
  File "/data/src/cockpit/test/testlib.py", line 223, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))), timeout=self.phantom_wait_timeout)
  File "/data/src/cockpit/test/testlib.py", line 647, in wait
    return self.run({'cmd': 'wait', 'cond': cond, 'timeout': timeout*1000})
  File "/data/src/cockpit/test/testlib.py", line 608, in run
    raise Error(res['error'])
Error: iframe.container-frame[name='10.111.111.16/shell/shell'] not found

----------------------------------------------------------------------
Ran 1 test in 96.751s
```

**Comment by stefwalter** [comment]
![check-multi-machine-key-testbasic-fail](https://cloud.githubusercontent.com/assets/795070/9625854/882ab524-5159-11e5-8451-55c9e278c1d9.png)

**Comment by petervo** [comment]
This was fixed with all the shell / index rework from #2756 and friends
